inspire/modules/deposit/workflows/literature.py

```diff
@@ -376,11 +376,11 @@ def remove_exclusive_fields(fieldlist):
             metadata['collections'].extend([{'primary': "arXiv"}, {'primary': "Citeable"}])
 
         if user_report_number:
+            report_numbers = [{'primary': v['report_number']} for v in user_report_number]
             if 'report_number' in metadata and metadata['report_number']:
```

Comment:
I re-map it in post processing so every report number generates its MARC correctly. ;)